### DISCUSSION TITLE
Import CSV file

### DISCUSSION THREAD

**Discussion by CEikermann** [trinodb/trino#18245] [Q&A]
Trino CLI can export data to CSV format using:
`trino --server http://localhost:8080 --progress -f query.sql --output-format CSV`

Is there also a way to import CSV back into the trino cluster via CLI?

Does anyone has a suggestion how to import a CSV file to an existing trino cluster without copying this file to S3 and register it as a table?

**Comment by nineinchnick** [comment]
You can register an external table in the Hive connector to read a CSV file stored in S3.

**Reply by CEikermann** [reply]
Yeah this could work, but I was looking for a simpler way via CLI to just import some data via CSV for testing purposes

Can you think of any alternative?

**Reply by nineinchnick** [reply]
You could use the [trino-storage](https://github.com/snowlift/trino-storage) connector to read a single file, it should be able to read local files but on the node, it can't transfer the file from where you run the CLI.

**Reply by hashhar** [reply]
To add more context. The clients are kept 'simple' which means they only support "querying" the system and don't offer client side data loading functionlity like Postgres's `\copy` or MySQL's client side `COPY`. And it's a non-goal to make the client be used for data ingestion.
